README.md

```diff
@@ -5,26 +5,22 @@ Preprocess search query by given patterns.
 # Pattern examples
 
 ```json5
-"re-search.patterns": [
-	{
-		// replace any `*` with regexp `.*` and activate regexp search
-		"*": {
-			"replacement": ".*",
-			"isRegex": true, // activate regexp search
-			"isCaseSensitive": false,
-			"matchWholeWord": false,
-		}
+"re-search.patterns": {
```

Comment:
Please keep array here, order of for-in for object is not guaranteed

Reply:
I changed this block because that was the structure I had to use for the extension to work.  Lines 48, 49 assume keys on an object.  `for..in` on arrays isn't a good idea for other reasons, but we could change the logic to use `Array.reduce` instead and then I can revert the config.

In practice the only scenarios that produce inconsistent key enumeration on JS objects is when you mutate them.  Deserializing the config will universally preserve the read order, and it makes for easier manual changes being one less layer deep - but if you'd prefer I can knock that out quickly.

Reply:
I didn't know what vscode expected for this structure here: https://github.com/profelis/re-search/blob/4ef0b318ac4e70289c7033da35c48b2c0955413f/package.json#L24-L26

But I presumed `examples` meant that you use one or the other - depending on how you feel about this issue I can make all three points consistent.

Reply:
Yes, please. I think array is more correct data structure for order-sensitive data (stop param introduce this sensitivity)

Reply:
Interesting, I used object here too, maybe nodejs keeps order and my information about order is outdated. I googled little this question and answer is still - it's depends. But now I'm confused and not still sure about this comment. Keep this changes

Reply:
Your knowledge is correct from the standards perspective for sure.  And other languages do exhibit sequencing issues from just reading a json blob.  But nodejs and every browser have kept consistent ordering as a practice since the early days of JS - so for configuration data it's pretty safe.